src/test/java/com/artipie/nuget/NugetITCase.java

```diff
@@ -0,0 +1,158 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.nuget;
+
+import com.amihaiemil.eoyaml.Yaml;
+import com.artipie.ArtipieServer;
+import com.jcabi.log.Logger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.StringContains;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+
+/**
+ * Integration tests for Nuget repository.
+ * @since 0.11
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+final class NugetITCase {
+
+    /**
+     * Temporary directory for all tests.
+     * @checkstyle VisibilityModifierCheck (3 lines)
+     */
+    @TempDir
+    Path tmp;
+
+    /**
+     * Tested Artipie server.
+     */
+    private ArtipieServer server;
+
+    /**
+     * Container.
+     */
+    private GenericContainer<?> cntn;
+
+    /**
+     * Artipie server port.
+     */
+    private int port;
+
+    @BeforeEach
+    void init() throws Exception {
+        final String name = "my-nuget";
+        this.server = new ArtipieServer(this.tmp, name, this.config());
+        this.port = this.server.start();
+        Testcontainers.exposeHostPorts(this.port);
+        Files.write(
+            this.tmp.resolve(String.format("repos/%s.yaml", name)),
+            this.config().getBytes()
+        );
+        this.nugetConfig();
+        this.cntn = new GenericContainer<>("mcr.microsoft.com/dotnet/sdk:5.0")
+            .withCommand("tail", "-f", "/dev/null")
+            .withWorkingDirectory("/home/")
+            .withFileSystemBind(this.tmp.toString(), "/home");
+        this.cntn.start();
+    }
+
+    @AfterEach
+    void tearDown() {
+        this.server.stop();
+        this.cntn.stop();
+    }
+
+    @Test
+    void shouldPushPackage() throws Exception {
+        MatcherAssert.assertThat(
+            this.pushPackage(),
+            new StringContains("Your package was pushed.")
+        );
+    }
+
+    private void nugetConfig() throws Exception {
```

Comment:
@genryxy this method does not return anything, but does an action: creates NuGet config file. It should be renamed to indicate that